strax/context.py

```diff
@@ -808,19 +818,64 @@ def get_iter(self, run_id: str,
                 max_messages=self.context_config['max_messages'],
                 timeout=self.context_config['timeout']).iter()
 
+        if progress_bar:
+            # Defining time ranges for the progress bar:
+            if time_range:
+                # user specified a time selection
+                start_time, end_time = time_range
+            else:
+                # If no selection is specified we have to get the last end_time:
+                start_time = 0 
+                end_time = float('inf')
+                for t in strax.to_str_tuple(targets):
+                    try:
+                        # Sometimes some metadata might be missing e.g. during tests.
+                        chunks = self.get_meta(run_id, t)['chunks']  
+                        start_time = max(start_time, chunks[0]['start'])
+                        end_time = min(end_time, chunks[-1]['end'])
+                    except (strax.DataNotAvailable, KeyError):
```

Comment:
> What if one of the chucks raises a KeyError, might you still be able to use the start and end_time of another target (as you loop over them)?

I actually had to add it since pytest would fail otherwise. 

> If so, you might want to set start_time, end_time = None, None above and check later that they aren't None anymore. If they are still None you can then set the progress_bar to False. 

Yes we can do that, but the progress bar may then be a bit odd. In those cases it could stop before 100% or go even further than 100%. Not sure if this would lead to any confusions.

Reply:
Okay I changed it such that if a metadata file of any target exists we will use that one.